dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/api/CommonSDKServlet.java

```diff
@@ -42,13 +46,23 @@ public class CommonSDKServlet extends FATServlet {
     @Test
     public void testClockNanoTime() throws InterruptedException {
         Clock clock = Clock.getDefault();
+        //get the clock start time
         long startNano = clock.nanoTime();
-        int sleepNano = 100000;
-        Thread.sleep(0, sleepNano);
+        //sleep for a known period
+        Duration sleepDuration = Duration.ofSeconds(1);
+        Thread.sleep(sleepDuration.toMillis());
+        //get the clock end time
         long endNano = clock.nanoTime();
-        long durationNano = endNano - startNano;
-        assertTrue(durationNano >= sleepNano);
-        assertTrue(durationNano < (10 * sleepNano));
+        //work out the duration according to the clock
+        long clockDurationNano = endNano - startNano;
+        Duration clockDuration = Duration.ofNanos(clockDurationNano);
+
+        //the duration according to the clock should be at least as long as the time we slept for (we might have slept longer)
+        assertTrue(clockDuration.compareTo(sleepDuration) >= 0);
+
+        //make sure that the clockDuration isn't vastly bigger than the intended sleep duration
+        Duration maxSleepDuration = sleepDuration.plus(Duration.ofSeconds(5));
+        assertTrue(clockDuration.compareTo(maxSleepDuration) < 0);
     }
```

Comment:
This is great, the clock tests look much more robust to me.

As an FYI, you can use the matchers `greaterThan()`, `lessThan()`, `between()` etc. to compare `Comparables` which should automatically get you reasonable test failure messages.